src/dbnode/persist/fs/commitlog/reader.go

```diff
@@ -423,106 +233,58 @@ func (r *reader) decodeAndHandleMetadata(
 	namespace.AppendAll(decoded.Namespace)
 
 	var (
+		idPool      = r.opts.IdentifierPool()
 		tags        ident.Tags
 		tagBytesLen = len(decoded.EncodedTags)
 	)
 	if tagBytesLen != 0 {
-		tagDecoderCheckedBytes.Reset(decoded.EncodedTags)
-		tagDecoder.Reset(tagDecoderCheckedBytes)
+		r.tagDecoderCheckedBytes.Reset(decoded.EncodedTags)
+		r.tagDecoder.Reset(r.tagDecoderCheckedBytes)
 
-		idPool := r.opts.IdentifierPool()
 		tags = idPool.Tags()
```

Comment:
True, it's not 100% clear whether they end up getting used or not ultimately because this is just the "reader".

I would probably prefer a caller copies to their own set of exact length tags, then returns the one that got pulled from the pool here back to the pool. 

In fact, maybe what I'll do is change it so we always reuse the ID and Tags between each read so that caller <has> to copy, which should reduce pool overuse in general for someone just wanting to read and not necessarily use these things for long term use (i.e. just quickly reading through the commit log).

Are you ok with that?

Reply:
So I think that has wider sweeping changes (changing to return something that needs to be copied) because it means we can't keep around the `seriesMetadata` lookup map in the commit log reader itself.

I'm going to keep it to just returning pooled things that can be returned if needed and a copy taken if needed, we can iterate on this in later changes perhaps.